app/src/main/java/com/github/bgabriel998/softwaredevproject/ProfileActivity.java

```diff
@@ -124,6 +132,72 @@ public void onComplete(Task<Void> task) {
                 });
     }
 
+    /**
+     * On add friend button click
+     * @param view
+     */
+    public void addFriendButton(View view) {
+        resetUI();
+        findViewById(R.id.submitFriendButton).setVisibility(View.VISIBLE);
+        findViewById(R.id.editTextFriend).setVisibility(View.VISIBLE);
+    }
+
+    /**
+     * On submit of add friend UI button click
+     * @param view
+     */
+    public void submitFriendButton(View view) {
+        String username = ((EditText)findViewById(R.id.editTextFriend)).getText().toString();
+        String currentUsername = account.getUsername();
+        Log.d("FRIENDS SIZE", "onSubmit: " + account.getFriends().size());
+
+        boolean found = false;
+        for(FriendItem friend: account.getFriends()) {
+            if(friend.hasUsername(username)) found = true;
+        }
+
+        // First, check if the username is valid, it's not the current user or it's already in the current user's friends
+        if(!Account.isValid(username) || username.equals(currentUsername) || found) {
+            Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), !Account.isValid(username) ? R.string.invalid_username : ( !found ? R.string.add_current_username : R.string.friend_already_added), Snackbar.LENGTH_LONG);
+            snackbar.show();
+            return;
+        }
+        checkUserExistsAndAdd(username);
+    }
+
+    /* Check that the user exists on the DB and, if so, add it to the friends of the current user  */
+    private void checkUserExistsAndAdd(String username) {
+        Database.refRoot.child(Database.CHILD_USERS).orderByChild(Database.CHILD_USERNAME).equalTo(username).addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(DataSnapshot snapshot) {
+                if (snapshot.exists()) {
+                    for (DataSnapshot user: snapshot.getChildren()) {
+                        String friendUid = user.getKey();
+                        Database.setChild(Database.CHILD_USERS + account.getId() + Database.CHILD_FRIENDS, Collections.singletonList(friendUid), Collections.singletonList(""));
+                        Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), R.string.friend_added, Snackbar.LENGTH_LONG);
+                        snackbar.show();
+                    }
+                }
+                else {
+                    Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), R.string.friend_not_present_db, Snackbar.LENGTH_LONG);
+                    snackbar.show();
+                }
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError error) {
+            }
+        });
+    }
+
+    /**
+     * On friends button click
+     * @param view
+     */
+    public void friendsButton(View view) {
+
+    }
```

Comment:
Remove method or add a TODO for later implementation
```suggestion
```

Reply:
TODO added.